nautilus_trader/adapters/binance/futures/schemas/market.py

```diff
@@ -173,18 +158,33 @@ class BinanceFuturesTradeData(msgspec.Struct):
     t: int  # Trade ID
     p: str  # Price
     q: str  # Quantity
-    X: BinanceFuturesOrderType  # Buyer order type
+    X: BinanceOrderType  # Buyer order type
```

Comment:
I gather you've combined the Spot and Futures orders by using all the `Optional`s, so then we can use a single order type for Binance?

Reply:
Yes, they both now use a single Enum. The variations between Spot and Futures order types was probably the single most frustrating difference between the APIs. Not only are they unnecessarily different, but they also share the *same* values for *different* order types... and the difference between the order types is inferred by the other values! :exploding_head:

Because of this, the `BinanceOrderType` enum has to have overriding parsing methods implemented in the child classes `BinanceFuturesEnumParser` & `BinanceSpotEnumParser`, even though it is common between Spot and Futures.

Many of the schemas were also consolidated just as you say, using `Optional`s. There's definitely a pattern with Binance where the core of each response is the same between Spot and Futures but the rarely used parameters differ arbitrarily /rant.